addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -1441,6 +1441,9 @@ export class Order extends PosModel {
             (this.pos_session_id % 10) * 100 +
             (this.sequence_number % 100)
         ).toString();
+        if (this.name.startsWith('Self-Order')) {
```

Comment:
It might be better to do this check in the pos_self_order module if possible. Because if only `point_of_sale` module is installed this will never be true. @vlst-odoo what do you think ?

Reply:
I am not sure but I don't think it is possible. We are gonna need to inherit most of the PoS views in the pos_self_order as this JS class is not linked to the order JS class in pos_self_order so whenever the ticket_screen for example gets rendered this is the JS setup() that is getting called.

This is actually what I discussed with @caburj

Reply:
I think you can do it. There is an Order patching already done in  the pos_store.js file in the pos_self_order module. There you just override the setup method and add this if statement